src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -76,4 +77,14 @@ public EditCommand parse(String args) throws ParseException {
         return Optional.of(ParserUtil.parseTags(tagSet));
     }
 
+    private Optional<Set<Address>> parseAddressesForEdit(Collection<String> addresses) throws IllegalValueException {
+        assert addresses != null;
+
+        if (addresses.isEmpty()) {
+            return Optional.empty();
+        }
+        Collection<String> addressSet = addresses.size() == 1 && addresses.contains("") ? Collections.emptySet() : addresses;
```

Comment:
Hmm, can't we just do an usual `if - else if - else` combining with the `if` statement above?